src/content/peerconnection/restart-ice/js/test.js

```diff
@@ -68,18 +64,16 @@ test('PeerConnection restart ICE sample', {skip: true}, t => {
     .then(() => seleniumHelpers.getStats(driver, 'pc1'))
     .then(stats => {
       firstStats = stats;
-      return driver.findElement(webdriver.By.id('restartButton')).click();
+      // listen for iceconnectionstatechange and store events.
+      return driver.executeScript(() => {
+        window.icestates = [];
+        pc1.addEventListener('iceconnectionstatechange', () => window.icestates.push(pc1.iceConnectionState));
+      });
     })
+    .then(() => driver.findElement(webdriver.By.id('restartButton')).click())
     .then(() => {
       t.pass('ICE restart triggered');
-      driver.manage().timeouts().setScriptTimeout(150000);
```

Comment:
I think this was flaky as it could add the event listener after the ice restart had gone to connected/completed. Adding the listener earlier and then checking for the checking->connected transition seems better.

Reply:
oh lovely. chrome doesn't go back to checking in this case. So this will have to listen for iegatheringstatechange instead...